standard-controller/src/main/java/io/enmasse/controller/standard/AddressController.java

```diff
@@ -745,30 +745,76 @@ private void checkAndRemoveDrainingBrokers(Set<Address> addresses) throws Except
+                            address.getStatus().setReady(false);
+                            address.getStatus().appendMessage("Address " + address.getSpec().getAddress() + " is not configured on broker " + brokerStatus.getContainerId());
+                        } else {
+                            ok++;
+                        }
+                        if (!queueNames.contains(address.getSpec().getAddress())) {
+                            address.getStatus().setReady(false);
+                            address.getStatus().appendMessage("Queue " + address.getSpec().getAddress() + " is not configured on broker " + brokerStatus.getContainerId());
+                        } else {
+                            ok++;
+                        }
+                    }
                     ok += RouterStatus.checkActiveAutoLink(address, routerStatusList);
                     ok += RouterStatus.checkForwarderLinks(address, routerStatusList);
                     break;
                 case "subscription":
+                    ok += checkBrokerStatus(brokerStatusCollector, address, clusterOk, clusterAddresses, clusterQueues);
+                    for (BrokerStatus brokerStatus : address.getStatus().getBrokerStatuses()) {
+                        Set<String> addressNames = clusterAddresses.get(brokerStatus.getClusterId());
+                        Set<String> queueNames = clusterQueues.get(brokerStatus.getClusterId());
+                        if (!addressNames.contains(address.getSpec().getTopic())) {
+                            address.getStatus().setReady(false);
+                            address.getStatus().appendMessage("Address " + address.getSpec().getTopic() + " is not configured on broker " + brokerStatus.getContainerId());
+                        } else {
+                            ok++;
+                        }
+                        if (!queueNames.contains(address.getSpec().getAddress())) {
+                            address.getStatus().setReady(false);
+                            address.getStatus().appendMessage("Queue " + address.getSpec().getAddress() + " is not configured on broker " + brokerStatus.getContainerId());
+                        } else {
+                            ok++;
+                        }
+                    }
                     ok += RouterStatus.checkForwarderLinks(address, routerStatusList);
                     break;
                 case "topic":
-                    ok += checkBrokerStatus(address, clusterOk);
+                    ok += checkBrokerStatus(brokerStatusCollector, address, clusterOk, clusterAddresses, clusterQueues);
                     for (RouterStatus routerStatus : routerStatusList) {
                         ok += routerStatus.checkLinkRoutes(address);
                     }
                     if (isPooled(addressPlan)) {
+                        for (BrokerStatus brokerStatus : address.getStatus().getBrokerStatuses()) {
```

Comment:
Why is this check only appropriate on the isPooled branch?

Reply:
For sharded topics, the addresses are part of the static broker.xml.

Reply:
> For sharded topics, the addresses are part of the static broker.xml.

I'd forgotten that difference.  Thanks.